src/config/tokens.ts

```diff
@@ -106,9 +120,34 @@ export function isNativeStableToken(tokenId: string) {
   return NativeStableTokenIds.includes(tokenId as TokenId)
 }
 
+export function isSwappable(token_1: string, token_2: string, chainId: number) {
+  const exchanges = MentoExchanges[chainId as ChainId]
+
+  if (!exchanges) return false
+
+  if (token_1 === token_2) return false
+
+  return exchanges.some(
+    (obj) =>
+      obj.assets.includes(getTokenAddress(token_1 as TokenId, chainId)) &&
+      obj.assets.includes(getTokenAddress(token_2 as TokenId, chainId))
+  )
+}
+
+export function getSwappableTokenOptions(token: string, chainId: ChainId) {
+  return getTokenOptionsByChainId(chainId)
+    .filter((tkn) => isSwappable(tkn, token, chainId))
+    .filter((tkn) => token !== tkn)
+}
+
 export function getTokenOptionsByChainId(chainId: ChainId): TokenId[] {
   const tokensForChain = TokenAddresses[chainId]
-  return tokensForChain ? (Object.keys(TokenAddresses[chainId]) as TokenId[]) : []
+
+  return tokensForChain
+    ? Object.entries(tokensForChain)
+        .filter(([, tokenAddress]) => tokenAddress !== '') // Allows incomplete 'TokenAddresses' list i.e When tokens are not on all chains
```

Comment:
we should probably not allow incomplete tokenAddresses right

Reply:
tokenAddresses (the token list which is used to populate the UI) isn't incomplete. The TokenAddress token to chain map above is incomplete (as there's no EUROC on main net). This is what allows us to move forward and add tokens for a chain without requiring it on all chains